code/modules/assembly/signaler.dm

```diff
@@ -6,147 +8,92 @@
 	materials = list(MAT_METAL=400, MAT_GLASS=120)
 	origin_tech = "magnets=1;bluespace=1"
 	wires = WIRE_RECEIVE | WIRE_PULSE | WIRE_RADIO_PULSE | WIRE_RADIO_RECEIVE
-
 	secured = TRUE
-	var/receiving = FALSE
-
 	bomb_name = "remote-control bomb"
-
+	/// Are we set to receieve a signal?
+	var/receiving = FALSE
+	/// Signal code
 	var/code = 30
+	/// Signal freqency itself
 	var/frequency = RSD_FREQ
-	var/delay = 0
-	var/datum/radio_frequency/radio_connection
-	var/airlock_wire = null
-
-/obj/item/assembly/signaler/New()
-	..()
-	if(SSradio)
-		set_frequency(frequency)
 
 /obj/item/assembly/signaler/Initialize()
-	..()
-	if(SSradio)
-		set_frequency(frequency)
+	. = ..()
+	GLOB.remote_signalers |= src
 
 /obj/item/assembly/signaler/Destroy()
-	if(SSradio)
-		SSradio.remove_object(src, frequency)
-	radio_connection = null
+	GLOB.remote_signalers -= src
 	return ..()
 
-/obj/item/assembly/signaler/describe()
-	return "[src]'s power light is [receiving ? "on" : "off"]"
+/obj/item/assembly/signaler/examine(mob/user)
+	. = ..()
+	. += "The power light is [receiving ? "on" : "off"]"
+
+// Called from activate(), actually invokes the signal on other signallers in the world
+/obj/item/assembly/signaler/proc/signal()
+	for(var/obj/item/assembly/signaler/S as anything in GLOB.remote_signalers)
+		if(S == src)
+			continue
+		if(S.receiving && (S.code == code) && (S.frequency == frequency))
+			S.signal_callback()
```

Comment:
To answer the "Why dont we check this on each signaler itself inside the `signal_callback()` proc?" question.

1. Overriding `signal_callback()` to just do the action as opposed to sanity checks is a lot easier to keep track of
2. By doing the checks here, it saves the time that would be jumping between procs and this loop and back, which can consume more CPU than you would think. 

Yes I am microoptimising this much.